src/common/dav/dav_internal.h

```diff
@@ -45,21 +47,40 @@ struct dav_phdr {
 			sizeof(struct stats_persistent)];
 };
 
+/*REVISIT*/
+struct bio_meta_instance;
+
 /* DAV object handle */
 typedef struct dav_obj {
 	char				*do_path;
 	uint64_t			 do_size;
 	void				*do_base;
 	struct palloc_heap		*do_heap;
 	struct dav_phdr			*do_phdr;
-	struct operation_context	*internal;
 	struct operation_context	*external;
 	struct operation_context	*undo;
-	struct dav_clogs		 clogs;
-	struct pmem_ops			 p_ops;	/* REVISIT */
+	struct mo_ops			 p_ops;	/* REVISIT */
 	struct stats			*do_stats;
 	int				 do_fd;
 	int				 nested_tx;
+	struct bio_meta_instance	*do_mi;
```

Comment:
The bio_meta_instance (actually name bio_meta_context in current code) is "umem_store->stor_priv", you don't have to store another pointer here.

Reply:
struct bio_meta_instance removed.